fiftyone/utils/torch.py

```diff
@@ -255,6 +256,124 @@ def ensure_torch_hub_requirements(
+
+    def __call__(self, sample):
+        return self.sample_dict_to_input(sample)
+
+    @property
+    def required_fields(self):
+        """a list of fields that are required in the sample for the GetItem to work."""
+        if not hasattr(self, "_required_fields_list"):
+            self._required_fields_list = ()
+        return self._required_fields_list
+
+    def add_required_fields(self, value):
+        """
+        Add fields to the required_fields list.
+        Args:
+            value: a list of fields or a single field to add to the required
+        """
+        if not hasattr(self, "_required_fields_list"):
+            self._required_fields_list = ()
+        if isinstance(value, str):
+            value = [value]
+        if not isinstance(value, list):
+            raise ValueError("required_fields must be a list or a string.")
+        self._required_fields_list = tuple(
+            set(list(self._required_fields_list) + value)
+        )
+        self.update_field_mapping()
+
+    @property
+    def field_mapping(self):
+        """
+        a dictionary mapping the fields in the sample to the input fields for the model
+        if not defined, we will assume the fields are the same
+        When this is updated, the behavior of __call__ should be updated accordingly.
+        This is what allows the user to specify the fields in the sample that are used as input to the model.
+        """
+        if not hasattr(self, "_field_mapping_dict"):
+            self._field_mapping_dict = {}
+        return self._field_mapping_dict
+
+    @field_mapping.setter
+    def field_mapping(self, value):
+        if not hasattr(self, "_field_mapping_dict"):
+            self._field_mapping_dict = {}
+        if value is None:  # generally on init
+            value = {k: k for k in self.required_fields}
+            for k, v in self._field_mapping_dict.items():
+                # if mixins have already set the field mapping, we should keep it
+                value[k] = v
+        if not isinstance(value, dict):
+            raise ValueError("field_mapping must be a dictionary.")
+        for k in value.keys():
+            if k not in self.required_fields:
+                raise ValueError(
+                    f"field_mapping key {k} not in required_fields."
+                )
+
+        for f in self.required_fields:
+            if f not in value:
+                value[f] = self.field_mapping.get(f, f)
+
+        self._field_mapping_dict = value
+
+    def update_field_mapping(self):
+        """
+        Update the field mapping for the get_item function.
+        """
+        for f in self.required_fields:
+            if f not in self.field_mapping:
+                self.field_mapping[f] = f
+
+    def validate_compatible_samples(self, samples):
+        """
+        Validate that the samples are compatible with the GetItem.
+        This should be called every time so we can fail quickly and loudly when needed.
```

Comment:
For performance reasons, I don’t think we should ever call validation on a per sample basis in the torch pipeline. If you must ensure fields, do it before assigning the samples by filtering only the ones with non null values when setting the sample ids